### PR TITLE
feat: resizing in DataTable

### PR DESCRIPTION
**Pull request by eunjae-lee**
## What does this PR do?

This adds column resizing functionality to DataTable, but enables only organization member list as a first start. (After a bit, we can have another PR to enable it for all DataTables)

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-4674 (Linear issue number - should be visible at the bottom of the GitHub issue description)

https://github.com/user-attachments/assets/70e70434-9ab7-4edb-b85d-4e3ef827e1ee

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

### CODE REVIEW

**Review by PeerRich — COMMENTED** [review]
nice!

File: packages/features/data-table/lib/resizing.ts
Comment (sean-brydon) [inline packages/features/data-table/lib/resizing.ts:1]: I have no issues with this but i wonder if its worth putting this into zustand with their persist plugin in the future

**Review by sean-brydon — DISMISSED** [review]
Tested - tried to break it as much as i can seems to be working perfectly!

One tiny NIT: We have a weird issue on our sticky columns (not blocking)
CleanShot 2024-12-04 at 10 11 44

File: packages/features/data-table/lib/resizing.ts
Comment (eunjae-lee) [reply packages/features/data-table/lib/resizing.ts:1]: I also had the same thought, but the logic was simple enough as-is. Btw, do we already have such plugin in place to persist data in local storage?

**Comment by eunjae-lee** [comment]
> Tested - tried to break it as much as i can seems to be working perfectly!
> 
> One tiny NIT: We have a weird issue on our sticky columns (not blocking) CleanShot 2024-12-04 at 10 11 44


Oh it happens when you resize it too small

https://github.com/user-attachments/assets/9b3d19b0-a343-4fa8-8b43-c7d538b949cc

changed it this:
Screenshot 2024-12-04 at 11 28 39

**Review by sean-brydon — APPROVED** [review]

**Pull request merged**